src/api/auth.hpp

```diff
@@ -33,6 +33,11 @@
 #include <common/optional.hpp>
 
 namespace mender {
+namespace update {
+namespace inventory {
+class InventoryAPI;
+}
+} // namespace update
```

Comment:
This should not be needed. `mender-update/inventory.hpp` should be `#include`d instead.

Reply:
I used a forward declaration here exactly to avoid using include, which can reduce compile time etc.

Reply:
Yeah, I got it. A nice habit for sure. But we use `#ifndef-#define` guards that should prevent those issues and we should be consistent.

Reply:
Include guards and forward declarations are not the same and serve different purposes. When you include a file the whole file will be inserted into the current one together with all the includes in that file as well (as long as they have not been included already because they have defined an include guard). Include guards prevent the same header file from being included multiple times and therefore prevent redefinition of the same symbol.

With forward declarations you get just the symbol with incomplete type. This can reduce compile time substantially, e.g. because you don't need to recompile several files when a header changes.